### PR TITLE
STRY55387793: Migrate Sonar Secret Authentication Changes to Dev Branch

### PR DESCRIPTION
**Pull request by deepakjanmanchi-sn**

### CODE REVIEW

File: .github/workflows/workflowToken.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  sonarAction:
+    needs: build
+    runs-on: ubuntu-latest
+    name: ServiceNow Devops Sonar
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: ServiceNow Devops Sonar
+        uses: ServiceNow/servicenow-devops-sonar@v2.0.0
```
Comment (sampath-methuku-servicenow) [inline .github/workflows/workflowToken.yml:35]: please update version to @3.1.0

File: README.md [hunk 2]
```diff
@@ -14,23 +14,48 @@ This custom action needs to be added at step level in a job to register sonar de
...
 ## Step 3: Configure the GitHub Action if need to adapt for your needs or workflows
+## For Token based Authentication which is available from @v2.0 , at ServiceNow instance
 ```yaml
 build:
     name: Build
     runs-on: ubuntu-latest
     steps:
       - name: ServiceNow DevOps Sonar Scan Results
-        uses: ServiceNow/servicenow-devops-sonar@v1
+        uses: ServiceNow/servicenow-devops-sonar@v2.0
```
Comment (sampath-methuku-servicenow) [inline README.md:41]: please update version to @3.1.0

File: README.md [hunk 3]
```diff
@@ -14,23 +14,48 @@ This custom action needs to be added at step level in a job to register sonar de
...
+          sonar-project-key: ${{ secrets.SONAR_PROJECT_KEY }}
+```
+## For Basic Authentication at ServiceNow instance
+```yaml
+build:
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+      - name: ServiceNow DevOps Sonar Scan Results
+        uses: ServiceNow/servicenow-devops-sonar@v1.34.2
```
Comment (sampath-methuku-servicenow) [inline README.md]: please update version to @3.1.0

File: .github/workflows/workflowToken.yml (hunk 1 above)
Comment (deepakjanmanchi-sn) [reply .github/workflows/workflowToken.yml]: updated

File: README.md (hunk 2 above)
Comment (deepakjanmanchi-sn) [reply README.md]: updated

**Review by sampath-methuku-servicenow — APPROVED** [review]

**Pull request merged**